ml_data_management_code_NB.Rmd

```diff
@@ -0,0 +1,342 @@
+---
+# ML Data Management Code
+
+## Intro
+
+This is the R Notebook for our ML Data Management Code. We originally worked on this in an RScript, but will now use an R Notebook for it. The data used in this file will come from a simulated dataset.
+
+## Download packages
+
+``` {r include = FALSE}
+library(tidyverse) # for data management
+library(psych) # to analyze data and descriptive statistics
+library(ggplot2) # to make some nice graphs
+library(pander) # to make some clean tables
+library(Rmisc) # to use summarySE() function
+library(gtsummary) # to make some cool summary tables that fit APA format
+
+```
+
+## Load in RScript where functions are created
+
+``` {r include = TRUE}
+
+#source(__.R)
+
+```
+
+## Download Simulated Data
+
+``` {r include = TRUE}
+
+source(simulate-data.R)
+
+```
+
+
+## Data Management
+
+### Merging data across time points
+
+#### Zach Method A 
+
+Built in studying the use case of needing to bind and merge different datasets that had different elements of missing data in each. This also works to bind datasets together.
+
+``` {r include = TRUE}
+
+# create dummy data
+x_pre <- data.frame(p = c(1, 2, 3), a = c(1, 2, 3), b = c(4, 5, 6), d = c(3, 4, 5))
+x_post <- data.frame(p = c(1, 2, 3), a = c(5, 2, 9), c = c(4, 5, 6))
+y_pre <- data.frame(p = c(4, 5, 6), b = c(1, 2, 3), c = c(4, 5, 6))
+y_post <- data.frame(p = c(4, 5, 6), b = c(1, 2, 3), d = c(4, 5, 6), e = c(1, 2, 3))
+
+
+# make function
+my_fun <- function(df){
+  df_name <- deparse(substitute(df))
+  if (grepl("pre", df_name)) {
+    colnames(df)[-1] <- paste(colnames(df)[-1], "t1", sep = "_") 
+  }
+  else if (grepl("post", df_name)) {
+    colnames(df)[-1] <- paste(colnames(df)[-1], "t2", sep = "_")
+  }
+  return(df)
+}
+
+# create a list of data frames
+my_list <- list(x_pre = x_pre, x_post = x_post,
+                y_pre = y_pre, y_post = y_post)
+
+# for loop through list
+for (i in seq_along(my_list)) {
+  df_name <- names(my_list)[i]
+  if (grepl("pre", df_name)) {
+    colnames(my_list[[i]])[-1] <- paste(colnames(my_list[[i]])[-1], "t1", sep = "_") 
+  }
+  else if (grepl("post", df_name)) {
+    colnames(my_list[[i]])[-1] <- paste(colnames(my_list[[i]])[-1], "t2", sep = "_")
+  }
+}
+
+```
+
+#### Zach Method B
+
+Alternative approach (probably better). Uses anonymous function and map
+
+``` {r include = TRUE}
+
+# re-create the initial list of data frames
+my_list <- list(x_pre = x_pre, x_post = x_post,
+                y_pre = y_pre, y_post = y_post)
+
+my_list <- Map(
+  function(list_df, list_names) {
+    if (endsWith(list_names, "pre")) {
+      names(list_df)[-1] <- paste(names(list_df)[-1], "t1", sep = "_")
+    } 
+    else if (endsWith(list_names, "post")) {
+      names(list_df)[-1] <- paste(names(list_df)[-1], "t2", sep = "_")
+    }
+    list_df
+  }, 
+  my_list, 
+  names(my_list)
+)
+
+#merge all data frames together
+final_product <- my_list %>% 
+  reduce(full_join)
+  
+```
+
+
+#### Matt Method (before Zach method)
```

Comment:
When I run my code and your code I get two different data frames. This suggests that there may be differences. I would only include your method if it is indeed doing the job. It is much simpler than whatever the hell I was doing!